beard/clustering/tests/test_blocking.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Beard.
+# Copyright (C) 2014 CERN.
+#
+# Beard is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""Test of blocking for clustering.
+
+.. codeauthor:: Gilles Louppe <g.louppe@cern.ch>
+
+"""
+from __future__ import division
+
+import numpy as np
+from numpy.testing import assert_equal
+from numpy.testing import assert_array_equal
+import pytest
+
+from sklearn.cluster import AgglomerativeClustering
+from sklearn.cluster import MiniBatchKMeans
+from sklearn.datasets import make_blobs
+from sklearn.metrics.pairwise import euclidean_distances
+from sklearn.utils import check_random_state
+
+from ..blocking import BlockClustering
+from ...metrics import paired_f_score
+
+
+def test_fit():
+    """Test fit."""
+    random_state = check_random_state(42)
+    X, y = make_blobs(centers=4, shuffle=False, random_state=random_state)
+
+    # Single block
+    clusterer = BlockClustering(
+        blocking="single",
+        base_estimator=AgglomerativeClustering(n_clusters=4,
+                                               linkage="complete"))
+    clusterer.fit(X)
+
+    assert_equal(len(clusterer.clusterers_), 1)
+    assert_equal(len(np.unique(clusterer.labels_)), 4)
+
+    # Precomputed blocks
+    clusterer = BlockClustering(
+        blocking="precomputed",
+        base_estimator=AgglomerativeClustering(n_clusters=2,
+                                               linkage="complete"))
+    clusterer.fit(X, blocks=(y <= 1))
+
+    assert_equal(len(clusterer.clusterers_), 2)
+    assert_equal(len(np.unique(clusterer.labels_)), 4)
+
+    # Custom blocking function
+    X_ids = np.arange(len(X)).reshape((-1, 1))
+
+    def _blocking(X_ids):
+        return y[X_ids.ravel()] <= 1  # block labels into {0,1} and {2,3}
+
+    def _distance(X_ids):
+        return euclidean_distances(X[X_ids.ravel()])
```

Comment:
The thing is that you dont know what will be the (index) values in `X_ids`, eg. if you do cross-validation `X_ids` will not contain all samples ids. (I agree though that the example here is a bit pointless, as it merely adds a level of indirection to the data, without doing anything clever.)